src/Settings.ts

```diff
@@ -78,9 +78,13 @@ export namespace Settings {
         }
     }
 
-    export namespace UserCustomSetting {
+    export namespace SettingsFile {
         export function path(): string | undefined {
-            return workspace.getConfiguration("maven.custom").get<string>("settingFileLocation");
+            let config : WorkspaceFolder | undefined ;
+            if (workspace.workspaceFolders !== undefined) {
+                config =  workspace.workspaceFolders[0];

```

Comment:
why always fetching the first workspace folder?

Reply:
the first entry corresponds to the value of  `rootPath`

Reply:
What if it's a multi-root workspace?

Reply:
Maybe use `workspace.workspaceFile` here would be better. At first I want this value changed when user select a folder that contains this setting, but I don't find how to get the folder that user current selected。